### PR TITLE
feat(gzip): support special * to match any type

### PR DESCRIPTION
**Pull request by RocFang**
### What this PR does / why we need it:

right now, type of types attr is array, and we dont't have a method to support the special `gzip_types *` config in Nginx Core. So maybe we can change type of types to support single string, and support * too.

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/plugins/gzip.lua [hunk 1]
```diff
@@ -21,17 +21,26 @@ local req_http_version = ngx.req.http_version
...
+                {
+                    type = "array",
+                    minItem = 1,
+                    items = {
+                        type = "string",
+                        minLength = 1,
+                    },
+                },
+                {
+                    type = "string",
```
Comment (spacewander) [inline apisix/plugins/gzip.lua:41]: I think `enum = {"*"}` is enough?

File: t/plugin/gzip.t [hunk 2]
```diff
@@ -436,6 +436,53 @@ POST /echo
 012345678
 --- more_headers
 Accept-Encoding: gzip
+Content-Type: text/html
```
Comment (spacewander) [inline t/plugin/gzip.t:439]: Better to use a special type to verify that any type can be matched.

File: apisix/plugins/gzip.lua (hunk 1 above)
Comment (RocFang) [reply apisix/plugins/gzip.lua:41]: @spacewander Yes, I think it's simpler to use {"*"}, this pr only try to follow nginx's config strictly.
If that not import, I will change it to {"*"}

Comment (RocFang) [reply apisix/plugins/gzip.lua:41]: misunderstood you firstly, now i got it.

**Review by spacewander — APPROVED** [review]

**Review by shuaijinchao — APPROVED** [review]

**Pull request merged**